### PR TITLE
Second screen

### PR DESCRIPTION
**Pull request by roppazvan**

### CODE REVIEW

File: src/app/store/models/tokenState.tsx [hunk 1]
```diff
@@ -105,6 +105,14 @@ export const tokenState = createModel<RootModel>()({
       ...state,
       activeTokenSet: data,
     }),
+    setUsedTokenSet: (state, data: UsedTokenSetsMap) => ({
```
Comment (six7) [inline src/app/store/models/tokenState.tsx:108]: Why was this changed?

File: src/types/SecondScreen.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { ThemeObject } from './ThemeObject';
+import { SingleToken } from './tokens';
+import { UsedTokenSetsMap } from './UsedTokenSetsMap';
+
+export type Token = SingleToken;
```
Comment (six7) [inline src/types/SecondScreen.ts:5]: Why not use SingleToken in the first place?

File: src/types/SecondScreen.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+export enum Clients {
+  PLUGIN = 'plugin',
+  SECOND_SCREEN = 'second-screen',
+}
+
+export interface TokenData {
+  id: string | number | null;
+  owner_email?: string;
+  created_at?: string;
+  ftData: {
```
Comment (six7) [inline src/types/SecondScreen.ts:18]: Can we change this? We are no longer called Figma Tokens

File: src/types/SecondScreen.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+export type TokenSets = Record<string, Token[]>;
+
+export enum Clients {
+  PLUGIN = 'plugin',
+  SECOND_SCREEN = 'second-screen',
+}
+
+export interface TokenData {
+  id: string | number | null;
+  owner_email?: string;
```
Comment (six7) [inline src/types/SecondScreen.ts:16]: When do we use snake_case and when camelCase?

File: src/types/Auth.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  expires_in: number;
+  refresh_token: string;
+  token_type: string;
+  user: UserData;
+}
+
+export interface AuthInfo {
+  email?: string;
+  password?: string;
+  repeatedPassword?: string;
```
Comment (six7) [inline src/types/Auth.ts:20]: Again feels weird to mix cases like that

File: src/context/AuthContext.tsx [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    setAuthInProgress(false);
+  }, []);
+
+  const signUp = useCallback(async (signUpInfo: AuthInfo) => {
+    setAuthInProgress(true);
+    const { data, error } = await supabase.signUp(signUpInfo);
+    if (error) {
+      setAuthError(error.msg || error.error_description);
+    } else {
+      handleLogin(data);
```
Comment (six7) [inline src/context/AuthContext.tsx:70]: Can you add mixpanel track() events for all these?

File: src/app/components/SecondScreen/index.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+  const secondScreenButton = (
+    <Box
+      css={{
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'space-between',
+        gap: '$3',
+        padding: '$2 $4',
+        background: isEnabled ? 'rgba(65, 63, 63, 1)' : 'transparent',
```
Comment (six7) [inline src/app/components/SecondScreen/index.tsx:38]: Can you use a token here? We need to support both light and dark mode, we cannot use rgb values

File: src/app/components/SecondScreen/index.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        borderRadius: '6px',
```
Comment (six7) [inline src/app/components/SecondScreen/index.tsx:39]:
```suggestion
        borderRadius: '$button',
```

File: src/app/components/AuthModal/index.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+  const [mode, setMode] = useState<AuthModes>(AuthModes.LOGIN);
+  const secondScreenEnabled = useSelector(secondScreenSelector);
+
+  const [values, setValues] = React.useState({
+    email: '',
+    password: '',
+  });
+
+  const handleChange = useCallback(
+    (e: any) => {
```
Comment (six7) [inline src/app/components/AuthModal/index.tsx:32]: Can we remove the any?

File: src/app/components/AuthModal/index.tsx [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+      </>
+    );
+  };
+
+  return (
+    <Modal close={dispatch.uiState.toggleSecondScreen} isOpen={secondScreenEnabled && !user}>
+      <Box css={{ display: 'flex', flexDirection: 'column', gap: '$5' }}>
+        <Input name="email" type="email" value={values.email} onChange={handleChange} label="Email" />
+        <Input name="password" type="password" value={values.password} onChange={handleChange} label="Password" />
+        <Button variant="primary" onClick={onSubmitButtonClick}>
```
Comment (six7) [inline src/app/components/AuthModal/index.tsx:86]: Can we still integrate a Forgot password link?

**Review by six7 — CHANGES_REQUESTED** [review]
Nice!

File: src/app/store/models/tokenState.tsx (hunk 1 above)
Comment (roppazvan) [reply src/app/store/models/tokenState.tsx:108]: I needed a way to update the `usedTokenSet` field in the state with the data from second-screen. And there was no way to do that in a straightforward way. same for themes.

File: src/types/SecondScreen.ts (hunk 4 above)
Comment (roppazvan) [reply src/types/SecondScreen.ts:16]: snake case is the recommended way for posgtres . Camel case is for the actual data inside ftData, I'll rename these to be more consistent
Screenshot 2023-02-03 at 10 17 11

File: src/types/Auth.ts (hunk 5 above)
Comment (roppazvan) [reply src/types/Auth.ts:20]: yeah, this is a mistake. theres no refresh_token there.

**Pull request merged**